easybuild/easyconfigs/o/OpenFold/OpenFold-1.0.0-foss-2021a-CUDA-11.3.1.eb

```diff
@@ -0,0 +1,94 @@
+easyblock = 'PythonBundle'
+
+name = 'OpenFold'
+version = '1.0.0'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://github.com/aqlaboratory/openfold'
+description = "A faithful PyTorch reproduction of DeepMind's AlphaFold 2"
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+builddependencies = [
+    # CMake is required to build ninja Python package (included as extension)
+    ('CMake', '3.20.1'),
+]
+
+dependencies = [
+    ('Python', '3.9.5'),
+    ('CUDA', '11.3.1', '', True),
+    ('SciPy-bundle', '2021.05'),
+    ('PyYAML', '5.4.1'),
+    ('Biopython', '1.79'),
+    ('HH-suite', '3.3.0'),
+    ('HMMER', '3.3.2'),
+    ('Kalign', '3.3.1'),
+    ('UCX-CUDA', '1.10.0', versionsuffix),
+    ('cuDNN', '8.2.1.32', versionsuffix, True),
+    ('NCCL', '2.10.3', versionsuffix),
+    ('dm-tree', '0.1.6'),
+    ('einops', '0.4.1'),
+    ('colossalai', '0.1.8', versionsuffix),
+    ('scikit-build', '0.11.1'),
+    # OpenFold also requires the AlphaFold modification to OpenMM
+    # https://github.com/aqlaboratory/openfold/blob/v1.0.0/scripts/install_third_party_dependencies.sh#L20-L24
+    # https://github.com/aqlaboratory/openfold/blob/v1.0.0/lib/openmm.patch
+    ('OpenMM', '7.5.1', '-DeepMind-patch'),
+]
+
+use_pip = True
+
+exts_list = [
+    ('PDBFixer', '1.7', {
+        'source_urls': ['https://github.com/openmm/pdbfixer/archive/refs/tags/'],
+        'sources': [{'download_filename': 'v%(version)s.tar.gz', 'filename': '%(name)s-%(version)s.tar.gz'}],
+        'checksums': ['a0bef3c52a7bbe69a6aea5333f51f3e7d158339be5829aed19b0344bd66d4eea'],
+    }),
+    ('ninja', '1.10.2.3', {
+        'checksums': ['e1b86ad50d4e681a7dbdff05fc23bb52cb773edb90bc428efba33fa027738408'],
+    }),
```

Comment:
does it need that specific version? We have `Ninja-1.10.1-GCCcore-10.2.0.eb`  which is this thing (to be used as a builddep only i assume?)

Reply:
but that provides only the `ninja` binary, and `deepspeed` requires the `ninja` python package...

Reply:
I guess we could enhance the (recent) `Ninja` easyconfigs to also install the Python bindings (I'm looking into that, PR coming soon...)

Reply:
see #16025

Reply:
It looks like the `ninja` Python package, provided by https://github.com/scikit-build/ninja-python-distributions, is actually a shim package, a very light-weight wrapper around the `ninja` binary so you can declare a dependency on it in `setup.py` & co...
With that in mind, I think it should be OK to just strip out the requirement for `ninja` as long as we provide the traditional `Ninja` as a (build) depemdency.

I'll look into this (and then close #16025 since that PR doesn't make much sense then)

Reply:
Hmm, just stripping out the requirement for the `ninja` Python package won't be a good idea, since `deepspeed` really does require it (if only to check whether `Ninja` is available), see https://github.com/microsoft/DeepSpeed/blob/316c4a43e0802a979951ee17f735daf77ea9780f/deepspeed/env_report.py#L54-L59

So unless we can somehow make the `ninja` Python package point to an existing `Ninja` installation rather than having it install it's own `ninja` binary, this may be a necessary evil that's hard to avoid... :-/

Reply:
I closed #16025, since that's clearly not the correct way forward.

Since the `ninja` Python package is also a *runtime* dependency for `deepspeed`, I don't see a better way out than the current approach being used here: install `ninja` as an extension in `OpenFold` (as opposed to trying to use the classic `Ninja` installation as a dependency somehow).